plugins/linux-v4l2/CMakeLists.txt

```diff
@@ -1,31 +1,43 @@
-project(linux-v4l2)
+cmake_minimum_required(VERSION 3.22...3.25)
+
+legacy_check()
 
 option(ENABLE_V4L2 "Build OBS with v4l2 support" ON)
+option(ENABLE_UDEV "Build linux-v4l2 with UDEV support" ON)
+
 if(NOT ENABLE_V4L2)
-  obs_status(DISABLED "linux-v4l2")
+  target_disable(linux-v4l2)
   return()
 endif()
 
-option(ENABLE_UDEV "Build linux-v4l2 with UDEV support" ON)
-
 find_package(Libv4l2 REQUIRED)
 find_package(FFmpeg REQUIRED COMPONENTS avcodec avutil avformat)
 
+if(OS_FREEBSD)
```

Comment:
This seems like it should be a finder for the kernel headers if they are required by us. Also where did the freebsd stanza come from? /usr/local/ implies non-managed garbage and seems very suspect.

Reply:
FreeBSD ports are always installed into `/usr/local` (because Unixes supply kernel binaries _and_ user land binaries, so `/usr` is the domain of FreeBSD and _not_ user-installed packages).

This means that `/usr/local/include` needs to be added to the include search paths used for `check_c_source_compiles` (because it does _not_ automatically inherit the include paths used for the _target_).